docs/apis-tools/java-client/authentication.md

```diff
@@ -0,0 +1,162 @@
+---
+id: authentication
+title: "Authentication"
+sidebar_label: "Authentication"
+description: "Authenticating the Java client."
+keywords: ["authentication", "username", "password", "basic", "oauth", "x509"]
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+There are different ways to authenticate to your cluster. The following sections describe how to authenticate using the Java client.
+
+<Tabs groupId="authenticationMethods" defaultValue="basic" queryString values={[{label: 'Basic', value: 'basic' },{label: 'OIDC', value: 'oidc' },{label: 'X.509', value: 'x509' }]}>
+
+<TabItem value="basic">
+
+In Java code, create a `BasicAuthCredentialsProvider` and provide it with your username and password.
+
+```java
+final var credentialsProvider =
+  new BasicAuthCredentialsProviderBuilder()
+    .applyEnvironmentOverrides(false)
```

Comment:
🔧  I wonder whether we actually should include it in this sample? Maybe we could just indicate with a comment that this is done to change the default to ensure these are always used?

Reply:
see comment in the Env Var section, I would propose to only raise this option there and explain it.